docs/docs/querying.md

```diff
@@ -284,22 +284,38 @@ something.findOne({
     // Will order by  otherfunction(`col1`, 12, 'lalala') DESC
     [sequelize.fn('otherfunction', sequelize.col('col1'), 12, 'lalala'), 'DESC'],
 
-    // Will order by name on an associated User
-    [User, 'name', 'DESC'],
+    // Will order an associated model's created_at using the model name as the association's name.
+    [Task, 'createdAt', 'DESC'],
 
-    // Will order by name on an associated User aliased as Friend
-    [{model: User, as: 'Friend'}, 'name', 'DESC'],
+    // Will order through an associated model's created_at using the model names as the associations' names.
+    [Task, Project, 'createdAt', 'DESC'],
 
-    // Will order by name on a nested associated Company of an associated User
-    [User, Company, 'name', 'DESC'],
+    // Will order by an associated model's created_at using the name of the association.
+    ['Task', 'createdAt', 'DESC'],
```

Comment:
The order query would look like;
```sql
ORDER BY "Task"."jsonField.level1"."level2" DESC
```

Simply put, there is a method `quoteIdentifiers` that takes a string with a "." in it, splits it two at the ".", and then wraps each piece of the string with identifier ticks.  It may actually be a good idea to deprecate this but it will cause confusion when upgrading from 3.